server/service/appconfig.go

```diff
@@ -590,6 +598,29 @@ func (svc *Service) validateMDM(
 				`Couldn't enable macos_setup.enable_end_user_authentication because no IdP is configured for MDM features.`)
 		}
 	}
+
+	updatingMacOSMigration := mdm.MacOSMigration.Enable != oldMdm.MacOSMigration.Enable ||
+		mdm.MacOSMigration.Mode != oldMdm.MacOSMigration.Mode ||
+		mdm.MacOSMigration.WebhookURL != oldMdm.MacOSMigration.WebhookURL
```

Comment:
It's already merged and it's a nit, but just FYI since the fields in `MacOSMigration` are all comparable, you can do `mdm.MacOSMigration != oldMdm.MacOSMigration`.